tests/test_settings.py

```diff
@@ -0,0 +1,99 @@
+"""Tests for the Settings class and module."""
+# Copyright 2018 Ian Stapleton Cordasco
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import unicode_literals
+
+from twine import exceptions
+from twine import settings
+
+import pretend
+import pytest
+
+
+def test_settings_takes_no_positional_arguments():
+    """Verify that the Settings initialization is kw-only."""
+    with pytest.raises(TypeError):
+        settings.Settings('a', 'b', 'c')
+
+
+def test_settings_transforms_config(monkeypatch):
+    """Verify that the settings object transforms the passed in options."""
+    replaced_get_repository_from_config = pretend.call_recorder(
```

Comment:
This amount of patching here makes me a little nervous. If we change any interfaces in `settings.utils` this test likely not fail, as the stubs here capture the interface at the time the tests were written. Can we accomplish this with less stubbing?

Reply:
:heavy_check_mark: Done